### ISSUE TITLE
Issue with  JObject property

### ISSUE THREAD

**Issue by greektreat** [microsoft/RulesEngine#233]
I am doing a POC with using this Rules Engine.  Our system will receive a event with some specific fields but also there is a payload field that can pass dynamic data it is stored as a JSON string inside a ms sql database. the data is converted to a class model where the Payload field is a Newtonsoft JObject
the string version of the payload is 
`"{\"Jurisdiction\":\"NY\", \"ContractStartDate\":\"2021-09-02T06:37:58.000\",\"Term\":30, \"Rate\":42.32}"`
```

var sherlockEvent = new Sherlock.Domain.Model.SherlockEvent
            {
                SourceSystem = sourceSystem,
                EventName = name,
                EventUniqueId = eventUniqueId,
                CorrelationId = correlationId,
                Timestamp = timestamp,
                DueDate = dueDate,
                ExpiryDate = expiryDate,
                UiUrl = uiUrl,
                DataUrl = dataUrl,
                Payload = JObject.Parse(payload)
            };
```
In the workflow rule expression I tried to do something simple. this expression didn't cause an error but returned as IsSuccess = false
`"Expression": "(sherlockEvent.Payload.Jurisdiction  == \"NY\") "`
What is strange was this should throw an exception. So in the ErrorMessage I tried to output the value
`"ErrorMessage": "Failed worflow $(sherlockEvent.Payload.Jurisdiction)",`

> "Failed worflow {\r\n  \"Jurisdiction\": \"NY\",\r\n  \"ContractStartDate\": \"2021-09-02T06:37:58\",\r\n  \"Term\": 30,\r\n  \"Rate\": 42.32\r\n}"
also if i reference it as $(sherlockEvent.Payload) it returns the same value.
it seems it is converting it into a string

Is there a better datatype to use for Payload? should it be a dynamic? or Dictionary<string,object>()?

**Comment by alexreich** [comment]
Combination of `dynamic` and `Dictionary<string, object>` will work. Here is a snippet from the [RE Editor](https://github.com/alexreich/RulesEngineEditor/blob/6640b15383aa2e8d6b453940bb42ec69dc518036/src/RulesEngineEditor/Pages/RulesEngineEditorPage.razor.cs#L151):

```csharp
List<InputRuleParameterDictionary> newInputs = new List<InputRuleParameterDictionary>();
WorkflowService.Inputs.ForEach(i =>
{
    InputRuleParameterDictionary newInput = new InputRuleParameterDictionary();
    newInput.InputRule = i.InputRule;
    newInput.Parameter = new Dictionary<string, object>();
    foreach (var p in i.Parameter)
    {
        try
        {
            newInput.Parameter.Add(p.Name, JsonConvert.DeserializeObject<dynamic>(p.Value, new ExpandoObjectConverter()));
        }
        catch (Exception ex)
        {
            inputJSONErrors += ex.Message + " ";
        }
    }
    newInputs.Add(newInput);
});
```

**Comment by abbasc52** [comment]
@greektreat Payload as Dictionary<string,object> should work.
